### PR TITLE
Add workaround for software style in ASA (sociology)

### PR DESCRIPTION
**Pull request by mikebader**
* Include condition for CSL to interpret an item with type=book and an entry for version as a computer program following: https://forums.zotero.org/discussion/comment/363537/#Comment_363537
* Make citation entry for computer programs when type=book and version is present following style:

Last, First. Year. Program Name (version #) [Computer program]. Retrieved from <url>.

### CODE REVIEW

**Comment by adam3smith** [comment]
(don't merge -- not quite right yet)

**Comment by adam3smith** [comment]
@mikebader -- where did you take the format 
> Last, First. Year. Program Name (version #) [Computer program]. Retrieved from .
from? 

I'm wondering in particular about the last part, which seems at odds with how ASA generally treats URLs, but I don't have the guide here & can't find anything on how ASA wants software cited online

**Comment by mikebader** [comment]
Thank you for taking this up!!! My ASA style guide, like everything from March 2020, has been hermetically sealed in my office on campus. I hope to get into my office soon, so I will confirm the style.

FWIW, I took the [APA style from the RIT website](https://library.rit.edu/citations/apa/computer-program-apa) because I seem to recall that ASA instructed authors to use APA style if something was not explicitly directed in the ASA style guide.

**Comment by mikebader** [comment]
@adam3smith Okay, I have banged my head against a wall. It turns out that I had an old version of the Style Guide. I purchased a new one, realized that it is based on the _Chicago Manual of Style_ not APA (doh!). But, neither the sixth edition of the ASA style guide nor the 17th Edition of CMOS has a reference for software specifically. The closest examples are Online Databases and Machine-Readable Data Files. 

The example for "research products from census and survey data" in the Online Databases section shows the example:

Ruggles, steven, Katie Genadek, Ronald Goeken, Josiah Grover, and Matthew Sobek. 2017. _Integrated Public Use Microdata Series: Version 7.0_ [dataset]. Minneapolis: University of Minnesota. doi:10.18128/D10.V7.0. 

The Machine-Readable Data Files Section includes the following note:

> Researchers frequently cite data and related information (coding keys, statistical program information, variable lists) available in Machine-Readable Data Files (MRDF). These sources may come from electronic media (e.g., CD-ROM, DVD) or be downloaded from an online source. For example, MRDF for the Youth Development Study are available under the "export metadata tab" at the following site:
>
> Mortimer, Jeylan T. 1998-2011. "Youth Development Study." [St. Paul, Minnesota], Ann Arbor, MI: Inter-university Consortium for Political and Social Research [distributor], 2015-12-18. doi:10.3886/ICPSR24881.v3. 

The IPUMS citation looks closest, but after checking [the website](https://doi.org/10.18128/D10.V7.0) the version number is actually part of the title in this specific case. In the other cases, the version number is just part of the DOI, which doesn't seem like it would generalize well. 

But you are absolutely correct that I botched the "Retrieved from" portion. I would defer to your expertise on this topic. Maybe something like:

Wheldon, Mark C. 2020. _CCMPP: Cohort Component Method of Population Projection_ [computer software], GitHub.com [distributor]. Version 0.0.0.9000. https://github.com/markalava/CCMPP/tree/7553c945f4e5856345eac7a662a7345bf79b0a68.

I feel like I've hit a wall on how to do this.

**Comment by adam3smith** [comment]
I agree that

> Wheldon, Mark C. 2020. _CCMPP: Cohort Component Method of Population Projection_ [computer software], GitHub.com [distributor]. Version 0.0.0.9000. https://github.com/markalava/CCMPP/tree/7553c945f4e5856345eac7a662a7345bf79b0a68.

Seems right (I'd be OK without the [distributor] part, too. When you're saying you hit a wall, do you mean you don't manage to do that? Any particular issue?

**Comment by mikebader** [comment]
Thanks! I meant that I was banging my head against a wall trying to find the proper format. I just pushed some code, following what you did before, that I _think_ makes the citation correct (I got rid of the distributor term like you suggested).

File: american-sociological-association.csl [hunk 1]
```diff
@@ -221,6 +231,20 @@
             <text macro="publisher"/>
           </group>
         </else-if>
+        <!--Software hack-->
+        <else-if type="book" variable="version" match="all">
+          <group prefix=" " delimiter=" ">
+            <group>
+              <text macro="title"/>
+              <text term="reference" prefix=" [" suffix="]." text-case="lowercase"/>
```
Comment (adam3smith) [inline american-sociological-association.csl:239]: Could you actually just hardcore computer program into the code and leave a Todo comment to localize once we have a proper term? That will travel better across languages than using a "wrong" redefined term.

**Review by adam3smith — COMMENTED** [review]
Looks great! One quick change request

**Comment by adam3smith** [comment]
Thanks!

**Pull request merged**